core/src/main/java/io/substrait/dsl/SubstraitBuilder.java

```diff
@@ -0,0 +1,288 @@
+package io.substrait.dsl;
+
+import com.github.bsideup.jabel.Desugar;
+import io.substrait.expression.AggregateFunctionInvocation;
+import io.substrait.expression.Expression;
+import io.substrait.expression.FieldReference;
+import io.substrait.expression.ImmutableFieldReference;
+import io.substrait.function.SimpleExtension;
+import io.substrait.plan.ImmutableRoot;
+import io.substrait.plan.Plan;
+import io.substrait.proto.AggregateFunction;
+import io.substrait.relation.Aggregate;
+import io.substrait.relation.Cross;
+import io.substrait.relation.Fetch;
+import io.substrait.relation.Filter;
+import io.substrait.relation.Join;
+import io.substrait.relation.NamedScan;
+import io.substrait.relation.Project;
+import io.substrait.relation.Rel;
+import io.substrait.relation.Set;
+import io.substrait.relation.Sort;
+import io.substrait.type.NamedStruct;
+import io.substrait.type.Type;
+import io.substrait.type.TypeCreator;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class SubstraitBuilder {
```

Comment:
See SubstraitRelNodeConverter for an example of how the methods in this class are used.

Reply:
I like the idea of moving towards an easier to write wrapper for generating plans, but any "usability" API would almost certainly need to come with proper docs, ideally containing short usage examples. An undocumented usability API, or one only documented by having to read the source code of other code, is going to be of limited use.

Reply:
This got merged before I circled back to hiding this.

Looking at https://substrait.io/spec/versioning/, because we're pre-1.0 I'm more comfortable with having it in. It is super useful for writing tests, both in substream-java and downstream.

> but any "usability" API would almost certainly need to come with proper docs, ideally containing short usage examples.

100% something to have in place before 1.0.